src/new/wallet.js

```diff
@@ -1125,6 +1402,34 @@ class HathorWallet extends EventEmitter {
 
     return balance;
   }
+
+  /**
+   * Return the addresses of the tx that belongs to this wallet
+   * The address might be in the input or output
+   * Removes duplicates
+   *
+   * @param {Object} tx Transaction data with array of inputs and outputs
+   *
+   * @return {Set} Set of strings with addresses
+   **/
+  getTxAddresses(tx) {
+    storage.setStore(this.store);
+    const walletAddresses = this.getAllAddresses();
```

Comment:
What's the type of `walletAddresses`? If it is a list, it might get really long and the searches inside the loops might be too expensive.

Reply:
I've changed to use another method `isAddressMine`, which accesses an object (where the keys are the addresses), then we can check if the address is from the wallet in O(1).

Reply:
Great. Thanks!